### DISCUSSION TITLE
Detecting change of  tab in tab control

### DISCUSSION THREAD

**Discussion by FraserElectronics** [AvaloniaUI/Avalonia#14790] [Q&A] [answered]
I am struggling to find any documentation on detecting when the user changes tab on a tab control.
I have tried binding the SelectionChanged event to my view model but get "Unable to find suitable setter or adder for property SelectionChanged" error. My handler looks like:

```
 public void OnTabChanged( object sender, SelectionChangedEventArgs e )
```

Could anyone point me in the right direction?

Andy

**Comment by stevemonaco** [comment, marked as answer]
Events are handled in code-behind, not your ViewModel. Normally, you bind `<TabControl SelectedItem="{Binding SelectedTab}" ... >` so that you can monitor changes to `SelectedTab` in your ViewModel. However, you could use the event in code-behind and forward a call into your ViewModel.

**Reply by FraserElectronics** [reply]
@stevemonaco Thanks for that 👍

Andy
